indexer/SymbolFormatter.cc

```diff
@@ -3,19 +3,123 @@
 #include <string_view>
 
 #include "absl/container/flat_hash_map.h"
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_replace.h"
 
 #include "clang/AST/Decl.h"
 #include "clang/Basic/SourceLocation.h"
 #include "clang/Basic/SourceManager.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "scip/scip.pb.h"
 
 #include "indexer/Enforce.h"
 #include "indexer/Hash.h"
 #include "indexer/LlvmAdapter.h"
 #include "indexer/SymbolFormatter.h"
 
+namespace {
+struct Escaped {
+  std::string_view text;
+};
+} // namespace
+
+namespace llvm {
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os, Escaped s) {
+  bool needsEscape = false;
+  bool needsBacktickEscape = false;
+  for (auto &c : s.text) {
+    if (('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z')
```

Comment:
Nice style. I LOVE when $a \leq x \leq b$ type expressions are written this way with the common operand in the middle, I wish everybody did this.